CMakeLists.txt

```diff
@@ -339,8 +342,18 @@ if(EMSCRIPTEN)
     add_compile_flag("-sDISABLE_EXCEPTION_CATCHING=0")
     add_link_flag("-sDISABLE_EXCEPTION_CATCHING=0")
   endif()
-  # make the tools immediately usable on Node.js
-  add_link_flag("-sNODERAWFS")
+  # make the tools immediately usable on Node.js, unless building for the browser
+  if (BUILD_FOR_BROWSER)
+    add_link_flag("-sENVIRONMENT=web,worker")
+    add_link_flag("-sINVOKE_RUN=0")
```

Comment:
Maybe add a comment about this flag? Something like "In the browser there is no natural place to provide commandline arguments for a commandline tool, so let the user run the main entry point themselves and pass in the arguments there."